[17.0][MIG] purchase_manual_delivery

StefanRijnhart opened the pull request:
* Implement setUpClass in tests
* Adapt to use stock.move quantity field in Odoo 17
* Fix default value if company of the order is not the user's default company
* Increase test coverage

Includes https://github.com/OCA/purchase-workflow/pull/2189, https://github.com/OCA/purchase-workflow/pull/2383

We're also proposing a small data model change. Previously, purchase order line's `existing_qty` included quantities from done stock moves. We're renaming the field to `qty_in_receipt` and changing the semantics to only include quanties from pending stock moves. The `pending_to_receive` will still be set under the same condition: if the ordered quantity is larger than the received quantity plus the quantities on pending stock moves. Data migration is included.

AdriaGForgeFlow committed:
[12.0][ADD] purchase_manual_delivery

[ADD] ability to change destination location in wizard

oca-travis committed:
[UPD] Update purchase_manual_delivery.pot

OCA-git-bot committed:
[UPD] README.rst

sysadminmatmoz committed:
Added translation using Weblate (Slovenian)

ChrisOForgeFlow committed:
[IMP] purchase_manual_delivery: black, isort, prettier

ChrisOForgeFlow committed:
[14.0][MIG] purchase_manual_delivery

ChrisOForgeFlow committed:
[ADD] configuration to make pickings on purchase orders

ChrisOForgeFlow committed:
[FIX] permissions

ChrisOForgeFlow committed:
IMP: added test for fields on purchase order line

ChrisOForgeFlow committed:
[FIX] rounding with decimal precision

ChrisOForgeFlow committed:
[FIX] added field invisible to ensure wizard works

oca-travis committed:
[UPD] Update purchase_manual_delivery.pot

OCA-git-bot committed:
[UPD] README.rst

ChrisOForgeFlow committed:
[14.0][IMP] added optional on field existing_qty

ChrisOForgeFlow committed:
[14.0][FIX] duplicate quantity on manual delivery

OCA-git-bot committed:
purchase_manual_delivery 14.0.1.0.1

sysadminmatmoz committed:
Translated using Weblate (Slovenian)

Currently translated at 100.0% (56 of 56 strings)

Translation: purchase-workflow-14.0/purchase-workflow-14.0-purchase_manual_delivery
Translate-URL: https://translation.odoo-community.org/projects/purchase-workflow-14-0/purchase-workflow-14-0-purchase_manual_delivery/sl/

PauBForgeFlow committed:
[MIG] purchase_manual_delivery: Migration to 16.0

PauBForgeFlow committed:
[FIX] correct partial demand in transfers

Commit:
[UPD] Update purchase_manual_delivery.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: purchase-workflow-16.0/purchase-workflow-16.0-purchase_manual_delivery
Translate-URL: https://translation.odoo-community.org/projects/purchase-workflow-16-0/purchase-workflow-16-0-purchase_manual_delivery/

ChrisOForgeFlow committed:
[16.0][FIX] purchase_manual_delivery, fix compute fields make both store True

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

Ivorra78 committed:
Added translation using Weblate (Spanish)

AlexPForgeFlow committed:
[FIX] purchase_manual_delivery: fix return on button_confirm_manual

OCA-git-bot committed:
[BOT] post-merge updates

astirpe committed:
[FIX] purchase_manual_delivery: manual delivery with double validation

Commit:
[UPD] Update purchase_manual_delivery.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: purchase-workflow-16.0/purchase-workflow-16.0-purchase_manual_delivery
Translate-URL: https://translation.odoo-community.org/projects/purchase-workflow-16-0/purchase-workflow-16-0-purchase_manual_delivery/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (65 of 65 strings)

Translation: purchase-workflow-16.0/purchase-workflow-16.0-purchase_manual_delivery
Translate-URL: https://translation.odoo-community.org/projects/purchase-workflow-16-0/purchase-workflow-16-0-purchase_manual_delivery/it/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (65 of 65 strings)

Translation: purchase-workflow-16.0/purchase-workflow-16.0-purchase_manual_delivery
Translate-URL: https://translation.odoo-community.org/projects/purchase-workflow-16-0/purchase-workflow-16-0-purchase_manual_delivery/es/

JordiMForgeFlow committed:
[IMP] purchase_manual_delivery: more specific menu name

Commit:
[UPD] Update purchase_manual_delivery.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: purchase-workflow-16.0/purchase-workflow-16.0-purchase_manual_delivery
Translate-URL: https://translation.odoo-community.org/projects/purchase-workflow-16-0/purchase-workflow-16-0-purchase_manual_delivery/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (65 of 65 strings)

Translation: purchase-workflow-16.0/purchase-workflow-16.0-purchase_manual_delivery
Translate-URL: https://translation.odoo-community.org/projects/purchase-workflow-16-0/purchase-workflow-16-0-purchase_manual_delivery/it/

ChristophAbenthungCibex committed:
[FIX]purchase_manual_delivery: qty compute fixes

Fixes the following 2 errors:
- When the product_uom_qty changes the recompute is not triggered
- In case that existing_qty > product_uom_qty the line should not be pending_to_receive

ChristophAbenthungCibex committed:
[FIX]purchase_manual_delivery: pending in forecast

Purchase Order Lines that are pending to receive are not considered for stock forecast. Thus, the manual and automatic reordering of products can be done despite having already create a Purchase Order and surpassing the max quantity. Confirmed Purchase Orders with pending to receive Purchase Order Lines are now considered at the forecast and the forecast report.

ChristophAbenthungCibex committed:
[FIX]purchase_manual_delivery: uom fixes

Fixed issues when uom_id and uom_po_id of a product are different.
- Comparison for Purchase Order Line pending to receive now uses the product quantity that uses the uom of the Purchase Order Line.
- When creating the Stock Move the quantity is now computed into the uom of the Stock Move

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
[BOT] post-merge updates

ntsirintanis committed:
[16.0][UPD] purchase_manual_delivery: do not show incoming shipment button when manual_delivery is False

OCA-git-bot committed:
[BOT] post-merge updates

StefanRijnhart committed:
[IMP] purchase_manual_delivery: pre-commit auto fixes

StefanRijnhart committed:
[MIG] purchase_manual_delivery: Migration to 17.0

StefanRijnhart committed:
[FIX] purchase_request: create new product for test

To prevent interference from other modules. In this case, there is a purchase
order from Odoo's demo data in a confirmed state but without a picking for the
product. The missing picking causes  purchase_manual_delivery to include the
PO line's quantity in the forecasted quantity of the product. This prevents
the procurement to go through in TestPurchaseRequestProcurement.test_orderpoint.

```
 2024-09-26 14:16:17,327 269 ERROR odoo odoo.addons.purchase_request.tests.test_purchase_request_procurement: FAIL: TestPurchaseRequestProcurement.test_orderpoint
Traceback (most recent call last):
  File "/__w/purchase-workflow/purchase-workflow/purchase_request/tests/test_purchase_request_procurement.py", line 81, in test_orderpoint
    self.assertEqual(
AssertionError: 0.0 != 5
```

astirpe reviewed (approved):
Thank you also for including #2189

StefanRijnhart commented:
/ocabot migration purchase_manual_delivery

StefanRijnhart committed:
[IMP] purchase_manual_delivery: rely on qty_received instead of 'done' stock moves

Rename existing_qty to qty_in_receipt. It now reflects the quantity for which there
are pending stock moves and no longer includes the quantities of 'done' stock moves.

Co-authored-by: Cas Vissers <cvi@360erp.com>

HviorForgeFlow reviewed (approved):
Code review :+1: Awesome job with the cherry-picks

HviorForgeFlow commented:
/ocabot merge nobump
